### PR TITLE
feat: ArgoCD support

### PR DESCRIPTION
**Pull request by Zebradil**
For every application, `myks` generates an ArgoCD Application resource with minimal viable configuration which can be changed via data values and overlays (in `_env/argocd` and `_apps/*/argocd`).

For every environment, `myks` generates an ArgoCD AppProject and Secret resources. The resulting resources can be changed via data values and overlays (in `_env/argocd` and `_apps/*/argocd`).

It is possible to set custom names and prefixes to the resources from data values files. The rest for now should be set via overlays.

### CODE REVIEW

File: internal/myks/environment.go [hunk 1]
```diff
@@ -210,6 +224,9 @@ func (e *Environment) saveRenderedEnvData(envDataYaml []byte) error {
 
 func (e *Environment) setEnvDataFromYaml(envDataYaml []byte) error {
 	var envDataStruct struct {
+		ArgoCD struct {
+			Enabled bool
```
Comment (kbudde) [inline internal/myks/environment.go:228]: Missing yaml struct annotations; but actually not needed to just lowercase yaml fields :) 

https://play.golang.com/p/zW7VNjuFkU-

**Review by kbudde — CHANGES_REQUESTED** [review]

File: internal/myks/assets/data-schema.ytt.yaml [hunk 2]
```diff
@@ -8,6 +8,48 @@
...
+  #! Namespace of the ArgoCD server.
+  namespace: argocd
+  app:
+    #! If not set, the name of the currently rendered application is used.
+    name: ""
+    #! Prefix of the ArgoCD application name.
+    prefix: ""
+    destination:
+      #! spec.destination.name of the ArgoCD application.
+      name: in-cluster
```
[the remark this answers is not in the document]
Comment (Zebradil) [reply internal/myks/assets/data-schema.ytt.yaml]: Good catch. Applied.

File: internal/myks/environment.go (hunk 1 above)
Comment (Zebradil) [reply internal/myks/environment.go]: I've added the anyways

**Review by kbudde — APPROVED** [review]

**Comment by kbudde** [comment]
LGTM.

nit: your PR summary is using `_app/*/argocd` instead of `_apps/*/argocd` ;)

**Pull request merged**